### PR TITLE
Allow verbosely printing Kafka statistics with a WITH option

### PR DESCRIPTION
**Pull request by umanwizard**
Creating a source with `verbose_stats_ms = <N>` will now cause a bunch of Kafka statistics to be pretty-printed to the log every N milliseconds. Mostly intended to be useful to us for debugging, so I haven't added it to docs/release notes.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/2650)

### CODE REVIEW

**Comment by umanwizard** [comment]
draft, not ready to land

**Comment by umanwizard** [comment]
Now ready to review.

File: src/dataflow/source/kafka.rs [hunk 1]
```diff
@@ -531,7 +530,11 @@ fn downgrade_capability(
 /// when the message queue switches from nonempty to empty.
 struct GlueConsumerContext(Mutex<SyncActivator>);
 
-impl ClientContext for GlueConsumerContext {}
+impl ClientContext for GlueConsumerContext {
+    fn stats(&self, statistics: Statistics) {
+        info!("Client stats: {:#?}", statistics);
+    }
```
Comment (benesch) [inline src/dataflow/source/kafka.rs:536]: Any particular reason to override the default implementation here?

Comment (umanwizard) [reply src/dataflow/source/kafka.rs:536]: {:#} instead of {}

**Pull request merged**